src/rgw/rgw_sync_module_pubsub.cc

```diff
@@ -1142,68 +1161,97 @@ class RGWPSFindBucketTopicsCR : public RGWCoroutine {
-                                          topics(_topics) {
-    owner = _owner;
-  }
+                                          topics(_topics),
+                                          owners({owner, rgw_user{}}),
+                                          has_subscriptions(false),
+                                          event_handled(false) {}
+
   int operate() override {
     reenter(this) {
       ldout(sync_env->cct, 10) << ": handle event: obj: z=" << sync_env->source_zone
                                << " event=" << json_str("event", *event, false)
                                << " owner=" << owner << dendl;
 
       ldout(sync_env->cct, 20) << "pubsub: " << topics->size() << " topics found for path" << dendl;
+      
+      if (topics->empty()) {
+        // if event has no topics - no further processing is needed
+        return set_cr_done();
+      }
+
+      if (perfcounter) perfcounter->inc(l_rgw_pubsub_event_triggered);
 
       for (titer = topics->begin(); titer != topics->end(); ++titer) {
-        ldout(sync_env->cct, 10) << ": notification for " << event->source << ": topic=" << (*titer)->name << ", has " << (*titer)->subs.size() << " subscriptions" << dendl;
+        ldout(sync_env->cct, 10) << ": notification for " << event->source << ": topic=" << 
+          (*titer)->name << ", has " << (*titer)->subs.size() << " subscriptions" << dendl;
 
         for (siter = (*titer)->subs.begin(); siter != (*titer)->subs.end(); ++siter) {
           ldout(sync_env->cct, 10) << ": subscription: " << *siter << dendl;
-
+          has_subscriptions = true;
+          sub_conf_found = false;
           for (oiter = owners.begin(); oiter != owners.end(); ++oiter) {
             /*
              * once for the global subscriptions, once for the user specific subscriptions
              */
             yield PSManager::call_get_subscription_cr(sync_env, env->manager, this, *oiter, *siter, &sub);
             if (retcode < 0) {
-              ldout(sync_env->cct, 10) << "ERROR: failed to find subscription config for subscription=" << *siter << " owner=" << *oiter << " ret=" << retcode << dendl;
+              last_sub_conf_error = retcode;
               continue;
             }
+            sub_conf_found = true;
 
             ldout(sync_env->cct, 20) << "storing event for subscription=" << *siter << " owner=" << *oiter << " ret=" << retcode << dendl;
             yield call(PSSubscription::store_event_cr(sync_env, sub, event));
             if (retcode < 0) {
               ldout(sync_env->cct, 10) << "ERROR: failed to store event for subscription=" << *siter << " ret=" << retcode << dendl;
-              continue;
+            } else {
+              event_handled = true;
+            }
+            ldout(sync_env->cct, 20) << "push event for subscription=" << *siter << " owner=" << *oiter << " ret=" << retcode << dendl;
+            yield call(PSSubscription::push_event_cr(sync_env, sub, event));
```

Comment:
maybe shouldn't call this unconditionally? we know at this point whether we're going to push anything, right?